worlds/cv64/__init__.py

```diff
@@ -0,0 +1,333 @@
+import os
+import typing
+import base64
+import threading
+import settings
+
+from BaseClasses import Item, Region, MultiWorld, Tutorial, ItemClassification
+from .items import CV64Item, filler_item_names, get_item_info, get_item_names_to_ids, get_item_counts
+from .locations import CV64Location, get_location_info, verify_locations, get_location_names_to_ids, base_id
+from .entrances import verify_entrances, get_warp_entrances, get_drac_rule
+from .options import CV64Options
+from .stages import get_stage_info, get_locations_from_stage, get_normal_stage_exits, vanilla_stage_order, \
+    shuffle_stages, generate_warps, get_region_names
+from .regions import get_region_info
+from .data import iname, lname, rname, ename
+from ..AutoWorld import WebWorld, World
+from .aesthetics import randomize_lighting, shuffle_sub_weapons, rom_empty_breakables_flags, rom_sub_weapon_flags,\
+    randomize_music, get_start_inventory_data, get_location_data, randomize_shop_prices, get_loading_zone_bytes,\
+    get_countdown_numbers
+from .rom import LocalRom, patch_rom, get_base_rom_path, get_item_text_color, CV64DeltaPatch
+from .client import Castlevania64Client
+
+
+class CV64Settings(settings.Group):
+    class RomFile(settings.UserFilePath):
+        """File name of the CV64 US 1.0 rom"""
+        copy_to = "Castlevania (USA).z64"
+        description = "CV64 (US 1.0) ROM File"
+        md5s = [CV64DeltaPatch.hash]
+
+    rom_file: RomFile = RomFile(RomFile.copy_to)
+
+
+class CV64Web(WebWorld):
+    theme = "stone"
+
+    setup_en = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Castlevania 64 randomizer connected to an Archipelago Multiworld.",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["Liquid Cat"]
+    )
+
+    tutorials = [setup_en]
+
+
+class CV64World(World):
+    """
+    Castlevania for the Nintendo 64 is the first 3D game in the franchise. As either whip-wielding Belmont descendant
+    Reinhardt Schneider or powerful sorceress Carrie Fernandez, brave many terrifying traps and foes as you make your
+    way to Dracula's chamber and stop his rule of terror.
+    """
+    game = "Castlevania 64"
+    item_name_groups = {
+        "Bomb": {iname.magical_nitro, iname.mandragora},
+        "Ingredient": {iname.magical_nitro, iname.mandragora},
+    }
+    location_name_groups = {stage: set(get_locations_from_stage(stage)) for stage in vanilla_stage_order}
+    options_dataclass = CV64Options
+    options: CV64Options
+    settings: typing.ClassVar[CV64Settings]
+    topology_present = True
```

Comment:
I was under the impression that this should be on if there are any entrances in a world instance that are not guaranteed to be static. For instance, Super Metroid turns it on when area rando is enabled and Timespinner always has it on probably specifically because of the Pyramid Key entrances being random every time. In the case of CV64, even on the most vanilla settings the exact warp destinations will be different every time, much like the Timespinner Pyramid Keys. So therefore, I was thinking it should be on for every world instance for a similar reason.

Reply:
Seems reasonable enough to me!